### DISCUSSION TITLE
DateEntry and TimeEntry not working (for me)

### DISCUSSION THREAD

**Discussion by Adsecularis** [ctreffe/alfred#230] [Alfred Q&A] [answered]
Had a functioning page and added to it
`self += al.DateEntry(
            leftlab=u'Datum',
            name="date"
        )
`
and got the error:
`AttributeError: module 'alfred3' has no attribute 'DateEntry'`

So I replaced it with a al.TextEntry.

Next I added
 `self += al.TimeEntry(
            leftlab=u'Uhrzeit',
            name="time",
            `
and got the same error:
`AttributeError: module 'alfred3' has no attribute 'TimeEntry'`

Are the elements DateEntry and TimeEntry currently not working in alfred3 or is my alfred3 version (2.3.1) outdated? 
Or could it be something else entirely?

**Comment by jobrachem** [comment, marked as answer]
Hi @Adsecularis, your alfred version is outdated. The classes your are looking for were added in v2.3.2: https://github.com/ctreffe/alfred/releases/tag/v2.3.2

You can simply run

```
pip install --upgrade alfred3
```

**Reply by Adsecularis** [reply]
The update process worked, but then I got error messages about missing modules. I tried some things which didn't work and then figured setting everything up again from the start might work.
So, I installed Python 3.8.10 (it's the latest version that runs under Windows 7. Yes, my PC is that old.)
Then made a new interpreter with Python 3.8 as it's base and newly installed the alfred3 package into that.

I tried to run the script and I got the error:
```
File "C:\Users\Adsecularis\PycharmProjects\Alfred3 Interpreter\lib\site-packages\alfred3\saving_agent.py", line 19, in <module>
    import mongomock
ModuleNotFoundError: No module named 'mongomock'
```
So I added mongomock to the interpreter and ran the script again.
And just like before all the upgrading I get:
`AttributeError: module 'alfred3' has no attribute 'DateEntry'`
The alfred version now is 2.3.3

What's wrong now?

**Reply by jobrachem** [reply]
Thanks for the report! The missing mongomock module is something that we need to add to `setup.py`.

It may be that `DateEntry` is not correctly exported. You can try to run:

```python
from alfred3.element.input import DateEntry
```

then use `DateEntry` instead of `al.DateEntry`.

**Reply by Adsecularis** [reply]
I did that for TimeEntry and DateEntry and it works :+1:

**Reply by jobrachem** [reply]
Thanks for letting me know! :)
